packages/dvilela/skills/memeooorr_abci/behaviour_classes/llm.py

```diff
@@ -270,10 +281,38 @@ def get_event(  # pylint: disable=too-many-locals,too-many-return-statements,too
                     None,
                     None,
                     None,
+                    current_timestamp,
                 )
 
-            # Fix amounts
             if action_name == "summon":
+                # Failsafe check: Recalculate or re-use time_since_last_summon
+
+                time_since_last_summon = current_timestamp - last_summon_timestamp
+                summon_cooldown_seconds = self.params.summon_cooldown_seconds
+
+                if time_since_last_summon < summon_cooldown_seconds:
+                    cooldown_remaining = (
+                        summon_cooldown_seconds - time_since_last_summon
+                    )
+                    self.context.logger.warning(
+                        f"LLM suggested 'summon' despite active cooldown ({cooldown_remaining:.2f}s remaining , going to retry)"
+                    )
```

Comment:
What does the message mean? Here we are not summoning, right? So what do we mean with retry?

Reply:
Isn't that overcomplicating it? Why dont you just remove the summon action from the available actions if we are still in the cooldown period?